test/test_intensify/test_successive_halving.py

```diff
@@ -0,0 +1,240 @@
+        self.config1 = Configuration(self.cs,
+                                     values={'a': 0, 'b': 100})
+        self.config2 = Configuration(self.cs,
+                                     values={'a': 100, 'b': 0})
+        self.config3 = Configuration(self.cs,
+                                     values={'a': 100, 'b': 100})
+
+        self.scen = Scenario({"cutoff_time": 2, 'cs': self.cs,
+                              "run_obj": 'runtime',
+                              "output_dir": ''})
+        self.stats = Stats(scenario=self.scen)
+        self.stats.start_timing()
+
+        self.logger = logging.getLogger(self.__module__ + "." + self.__class__.__name__)
+
+    def test_top_k(self):
+        """
+            test _top_k()
+        """
+        intensifier = SuccessiveHalving(
+            tae_runner=None, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+            rng=np.random.RandomState(12345),
+            instances=[1], min_budget=1)
+        self.rh.add(config=self.config1, cost=1, time=1,
+                    status=StatusType.SUCCESS, instance_id=1,
+                    seed=None,
+                    additional_info=None)
+        self.rh.add(config=self.config2, cost=2, time=2,
+                    status=StatusType.SUCCESS, instance_id=2,
+                    seed=None,
+                    additional_info=None)
+        self.rh.add(config=self.config3, cost=10, time=10,
+                    status=StatusType.SUCCESS, instance_id=3,
+                    seed=None,
+                    additional_info=None)
+        conf = intensifier._top_k(configs=[self.config2, self.config1, self.config3],
+                                  k=1, run_history=self.rh)
+
+        self.assertEqual(conf, [self.config1])
+
+    @attr('slow')
+    def test_intensify_1(self):
+        """
+           test intensify with quality objective & cutoff as budget
+        """
+
+        def target(x, instance, seed, budget):
+            return (x['a'] + 1 + budget) / 1000.
+
+        taf = ExecuteTAFuncDict(ta=target, stats=self.stats, run_obj='quality')
+        taf.runhistory = self.rh
+
+        intensifier = SuccessiveHalving(
+            tae_runner=taf, stats=self.stats,
+            traj_logger=TrajLogger(output_dir=None, stats=self.stats),
+            rng=np.random.RandomState(12345), deterministic=True, run_obj_time=False,
+            instances=[1], min_budget=1, max_budget=3, eta=2)
+
+        self.rh.add(config=self.config1, cost=1, time=1,
+                    status=StatusType.SUCCESS, instance_id=1,
+                    seed=0, budget=3,
+                    additional_info=None)
+
+        inc, _ = intensifier.intensify(challengers=[self.config2],
+                                       incumbent=self.config1,
+                                       run_history=self.rh,
+                                       aggregate_func=average_cost)
+
+        self.assertEqual(inc, self.config2)
```

Comment:
Could you please change all these tests to either check the full run history or mock out the comparison function and check the call arguments?